client/src/pages/CreateSpotForm.js

```diff
@@ -1,3 +1,135 @@
+import styled from 'styled-components';
+import {useState} from 'react';
+
 export default function CreateSpotForm() {
-  return <h1>CreateSpot</h1>;
+  const initialPersonalSpotState = {
+    username: '',
+    email: '',
+    further_info: '',
+    meet_others: true,
+    latitude: '',
+    longtitude: '',
+  };
+
+  const [lat, setLat] = useState(null);
+  const [lng, setLng] = useState(null);
+  const [status, setStatus] = useState(null);
+  const [personalSpot, setPersonalSpot] = useState(initialPersonalSpotState);
+
+  function getLocation() {
+    if (!navigator.geolocation) {
+      setStatus('Geolocation is not supported by your browser');
+    } else {
+      setStatus('Locating...');
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          setStatus(null);
+          setLat(position.coords.latitude);
+          setLng(position.coords.longitude);
+        },
+        () => {
+          setStatus('Unable to retrieve your location');
+        }
+      );
+    }
+  }
+
+  function handleFormSubmit(event) {
```

Comment:
I'm not quite sure what you're trying to do on submit? Shouldn't the data be send somewhere? Please review.